Mathlib/Topology/Constructions.lean

```diff
@@ -706,11 +706,27 @@ theorem continuousAt_pi' {f : X → ∀ i, A i} {x : X} (hf : ∀ i, ContinuousA
   continuousAt_pi.2 hf
 
 @[fun_prop]
-protected theorem ContinuousAt.piMap {Y : ι → Type*} [∀ i, TopologicalSpace (Y i)]
-    {f : ∀ i, A i → Y i} {x : ∀ i, A i} (hf : ∀ i, ContinuousAt (f i) (x i)) :
+protected theorem ContinuousAt.piMap {f : ∀ i, A i → B i} {x : ∀ i, A i}
+    (hf : ∀ i, ContinuousAt (f i) (x i)) :
     ContinuousAt (Pi.map f) x :=
   continuousAt_pi.2 fun i ↦ (hf i).comp (continuousAt_apply i x)
 
+protected lemma Topology.IsInducing.piMap {f : ∀ i, A i → B i}
+    (hf : ∀ i, IsInducing (f i)) : IsInducing (Pi.map f) := by
+  refine isInducing_iff_nhds.mpr fun x ↦ le_antisymm ?_ (fun s ↦ ?_)
+  · simpa [← Filter.map_le_iff_le_comap]
+      using (Continuous.piMap fun i ↦ (hf i).continuous).continuousAt
+  · simp only [Filter.mem_comap, nhds_pi, Filter.mem_pi']
+    rintro ⟨I, t, ht, ht'⟩
+    simp only [fun i ↦ Topology.isInducing_iff_nhds.mp (hf i) (x i),
+      Filter.mem_comap] at ht
+    choose u hu hu' using ht
+    exact ⟨Set.pi I u, ⟨I, u, hu, le_rfl⟩, subset_trans (fun v ↦ by aesop) ht'⟩
```

Comment:
I suggest adding the following right after [Filter.tendsto_piMap_pi](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Pi.html#Filter.tendsto_piMap_pi)
```lean
theorem pi_comap {β : ι → Type*} {f : ∀ i, α i → β i} {l : ∀ i, Filter (β i)} :
    pi (fun i ↦ comap (f i) (l i)) = comap (Pi.map f) (pi l) := by
  simp [Filter.pi, Filter.comap_comap]
  rfl
```
Then you can do:
```suggestion
protected lemma Topology.IsInducing.piMap {f : ∀ i, A i → B i}
    (hf : ∀ i, IsInducing (f i)) : IsInducing (Pi.map f) := by
  simp [isInducing_iff_nhds, nhds_pi, (hf _).nhds_eq_comap, Filter.pi_comap]
```